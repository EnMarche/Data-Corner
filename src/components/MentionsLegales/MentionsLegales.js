import { makeStyles } from '@material-ui/core'

const useStyles = makeStyles(theme => ({
  mentionsLegales: {
    color: theme.palette.mentionsLegales,
    fontSize: '10px',
    position: 'absolute',
    bottom: theme.spacing(2),
  },
  version: {
    margin: theme.spacing(0, 0, 1.25, 4),
  },
  list: {
    paddingLeft: theme.spacing(4),
    '& a': {
      fontWeight: 400,
    },
  },
  signature: {
    margin: theme.spacing(0, 4),
  },
}))

const messages = {
  title: "Je m'engage",
}

function MentionsLegales() {
  const classes = useStyles()

  return (
    <div className={classes.mentionsLegales}>
      <div className={classes.version}>
        {messages.title}@<strong>{process.env.REACT_APP_VERSION}</strong>
      </div>
      <ul className={classes.list}>
        <li>
          <a href="https://donnees.en-marche.fr/">Mes données personnelles</a>
        </li>
        <li>
          <a href="https://en-marche.fr/mentions-legales">Mentions légales</a>
        </li>
        <li>
          <a href="https://en-marche.fr/politique-cookies">Politique de cookies</a>
        </li>
        <li>
          <a href="https://en-marche.fr/politique-protection-donnees">Politique de protection des données</a>
        </li>
        <li>
          <a href="https://www.bkms-system.com/bkwebanon/report/clientInfo?cin=Jp3wHD&c=-1&language=fre">
            Cellule d&apos;alerte
          </a>
        </li>
      </ul>
      <div className={classes.signature}>Designé et assemblé par le Pôle Tech & Innovation</div>
    </div>
  )
}

export default MentionsLegales
