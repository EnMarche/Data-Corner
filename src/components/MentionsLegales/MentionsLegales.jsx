import React from 'react';

function MentionsLegales() {
    return (
        <div className="mentionsLegales">
            <div className="version">DataCorner@<strong>{process.env.REACT_APP_VERSION}</strong></div>

            <ul className="mentionLegalesList">
                <li>
                    <a href="https://donnees.en-marche.fr/">
                        Mes données personnelles
                    </a>
                </li>
                <li>
                    <a href="https://en-marche.fr/mentions-legales">
                        Mentions légales
                    </a>
                </li>
                <li>
                    <a href="https://en-marche.fr/politique-cookies">
                        Politique de cookies
                    </a>
                </li>
                <li>
                    <a href="https://en-marche.fr/politique-protection-donnees">
                        Politique de protection des données
                    </a>
                </li>
                <li>
                    <a href="https://www.bkms-system.com/bkwebanon/report/clientInfo?cin=Jp3wHD&c=-1&language=fre">
                        Cellule d&apos;alerte
                    </a>
                </li>
            </ul>
        </div>
    );
}

export default MentionsLegales;
