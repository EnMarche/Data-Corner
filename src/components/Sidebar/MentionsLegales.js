import { useState } from 'react'
import { styled } from '@mui/system'
import { Menu, MenuItem as MuiMenuItem } from '@mui/material'
import ErrorOutlineRoundedIcon from '@mui/icons-material/ErrorOutlineRounded'

const LegalNoticesWrapper = styled('span')(
  ({ theme }) => `
  padding: ${theme.spacing(0.2, 1, 0)};
  margin-right: 10px;
  background: ${theme.palette.menu.background.hover};
  border-radius: 30px;
`
)

const LegalNoticesIcon = styled(ErrorOutlineRoundedIcon)`
  font-size: 14px;
  color: ${({ theme }) => theme.palette.menu.color.main};
  cursor: pointer;
`

const MenuItem = styled(MuiMenuItem)(
  ({ theme }) => `
  color: ${theme.palette.menu.color.main};
  font-size: 10px;
  display: flex;
  margin: ${theme.spacing(0.5, 2)};
  padding: ${theme.spacing(0.5, 1)};
  border-radius: 6px;
  &:hover {
    color: ${theme.palette.menu.color.main};
    background: ${theme.palette.menu.background.hover};
  }
`
)

const messages = {
  personalData: 'Mes données personnelles',
  legalNotices: 'Mentions légales',
  cookiesPolicy: 'Politique de cookies',
  dataProtection: 'Politique de protection des données',
  warning: "Cellule d'alerte",
  signature: 'Designé et assemblé par le Pôle Tech & Innovation',
}

const MentionsLegales = () => {
  const [anchorEl, setAnchorEl] = useState(null)
  const open = Boolean(anchorEl)

  const handleClick = event => {
    setAnchorEl(event.currentTarget)
  }

  const handleClose = () => {
    setAnchorEl(null)
  }

  return (
    <LegalNoticesWrapper>
      <LegalNoticesIcon onClick={handleClick} />
      <Menu
        anchorEl={anchorEl}
        open={open}
        onClose={handleClose}
        anchorOrigin={{
          vertical: 'center',
          horizontal: 'right',
        }}
        transformOrigin={{
          vertical: 'center',
          horizontal: 'left',
        }}
        sx={{
          '& .MuiMenu-paper': {
            bgcolor: 'menu.background.main',
          },
        }}
      >
        <MenuItem onClick={handleClose}>
          <a href="https://donnees.en-marche.fr/">{messages.personalData}</a>
        </MenuItem>
        <MenuItem onClick={handleClose}>
          <a href="https://en-marche.fr/mentions-legales">{messages.legalNotices}</a>
        </MenuItem>
        <MenuItem onClick={handleClose}>
          <a href="https://en-marche.fr/politique-cookies">{messages.cookiesPolicy}</a>
        </MenuItem>
        <MenuItem onClick={handleClose}>
          <a href="https://en-marche.fr/politique-protection-donnees">{messages.dataProtection}</a>
        </MenuItem>
        <MenuItem onClick={handleClose}>
          <a href="https://www.bkms-system.com/bkwebanon/report/clientInfo?cin=Jp3wHD&c=-1&language=fre">
            {messages.warning}
          </a>
        </MenuItem>
      </Menu>
    </LegalNoticesWrapper>
  )
}

export default MentionsLegales
