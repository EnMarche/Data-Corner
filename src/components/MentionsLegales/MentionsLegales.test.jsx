
import { render } from '@testing-library/react';
import MentionsLegales from './MentionsLegales';

describe('MentionsLegales', () => {
    it('renders', () => {
        const { container } = render(<MentionsLegales />);

        expect(container).toMatchSnapshot();
    });
});
