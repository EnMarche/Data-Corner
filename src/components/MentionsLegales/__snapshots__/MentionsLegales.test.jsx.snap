// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentionsLegales renders 1`] = `
<div>
  <div
    class="mentionsLegales"
  >
    <div
      class="version"
    >
      DataCorner@
      <strong />
    </div>
    <ul
      class="mentionLegalesList"
    >
      <li>
        <a
          href="https://donnees.en-marche.fr/"
        >
          Mes données personnelles
        </a>
      </li>
      <li>
        <a
          href="https://en-marche.fr/mentions-legales"
        >
          Mentions légales
        </a>
      </li>
      <li>
        <a
          href="https://en-marche.fr/politique-cookies"
        >
          Politique de cookies
        </a>
      </li>
      <li>
        <a
          href="https://en-marche.fr/politique-protection-donnees"
        >
          Politique de protection des données
        </a>
      </li>
      <li>
        <a
          href="https://www.bkms-system.com/bkwebanon/report/clientInfo?cin=Jp3wHD&c=-1&language=fre"
        >
          Cellule d'alerte
        </a>
      </li>
    </ul>
    <div
      class="signature"
    >
      Designé et assemblé par le Pôle Tech & Innovation
    </div>
  </div>
</div>
`;
